### DISCUSSION TITLE
Accessing information in data object

### DISCUSSION THREAD

**Discussion by psheehymn** [gapitio/gapit-htmlgraphics-panel#173] [Q&A] [answered]
I am running into an odd issue with a type of query that I have been using extensively to this point. I need to use a variation on the query syntax to obtain the information and I am not getting back the same structure. I also cannot figure out the correct syntax to access the data I need.

In the original version of the query, one of the elements of the object I get back is a **name** field which I can easily access as _data.series[i].fields[1].name_.
![name](https://github.com/gapitio/gapit-htmlgraphics-panel/assets/93273674/ad7eaef0-4f35-42a4-bb30-6ec6836e763a)

In the new variation of the query, I do not get a name field for some reason (the syntax of the query is the same; the container I am querying from is slightly different because the data relates to a mobile application as opposed to a web application). However, I do see the data I need inside the **labels** object.
![viewname](https://github.com/gapitio/gapit-htmlgraphics-panel/assets/93273674/9bdca092-24f5-45f9-8ada-52d90ab8a80d)

What I can't figure out is the syntax to address the field inside **labels**. _data.series[i].fields[1].labels.view_ gives me back a value of "undefined." Any other variation I tried, including _data.series[i].fields[1].labels.view.name_, or trying to use the listed object prototypes, or several attempted variations on array indexing syntax all give errors similar to "Cannot read properties of undefined (reading 'valueOf')."

Do you have any insight on the proper syntax to access the value inside of the _view.name_ inside the **labels** object?

**Comment by ZuperZee** [comment, marked as answer]
The view.name is the key.
Changing data.series[i].fields[1].labels.view.name to data.series[i].fields[1].labels["view.name"] should fix it.

**Reply by psheehymn** [reply]
Now I feel a bit sheepish, I didn't think of that variant; that was it. Thank you!!
